### PR TITLE
Feat/game FUT-18

### PR DESCRIPTION
**Pull request by bkchilidawg**
Ok, We are merging feat/game to main. Good luck everyone

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -1 +1,131 @@
...
+    new(games_data, teams_data, game_teams_data)
+  end
+
+  def initialize(games_data, teams_data, game_teams_data)
+    @games_data = games_data
+    @teams_data = teams_data
+    @game_teams_data = game_teams_data
+  end
+
+  def highest_total_score
```
Comment (trpubz) [inline lib/stat_tracker.rb:20]: looks like similar code in both #highest_total_score and #lowest_total_score; opportunity to refactor in iteration 4. 
Perhaps get all total scores into array and then call #max_by and #min_by to find highest and lowest

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -1 +1,135 @@
...
+        home_wins += 1
+      end
+    end
+
+    if games != 0
+      (home_wins.to_f / games * 100.0).round(1)
+    end
+  end
+
+  def percentage_visitor_wins
```
Comment (trpubz) [inline lib/stat_tracker.rb:60]: this should do no logic other than call `self.percentage_home_wins + self.percentage_ties` and then subtract from 100

**Review by trpubz — CHANGES_REQUESTED** [review]

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -1 +1,135 @@
...
+    @games_data.each do |game|
+      total_score = (game[:home_goals].to_i + game[:away_goals].to_i)
+
+      lowest_score = total_score if total_score < lowest_score
+      lowest_score.round(1)
+    end
+    lowest_score
+  end
+
+  def percentage_home_wins
```
[the remark this answers is not in the document]
Comment (kameronk92) [reply lib/stat_tracker.rb:45]: Looked at using @game_teams_data, and like this idea, however it brings up a new problem. Every game is on two rows in the @game_teams_data (one row for each team), so if we have an incomplete data set (only one team out of 2 reporting) the math will be off.

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -1 +1,135 @@
...
+        visitor_wins += 1
+      end
+    end
+
+    if games != 0
+      (visitor_wins.to_f / games * 100.0).round(1)
+    end
+  end
+
+  def percentage_ties
```
[the remark this answers is not in the document]
Comment (kameronk92) [reply lib/stat_tracker.rb:75]: Same issue comes up if game_teams_data is incomplete

File: lib/stat_tracker.rb (hunk 1 above)
Comment (bkchilidawg) [reply lib/stat_tracker.rb]: closed with helper

File: lib/stat_tracker.rb [hunk 5]
```diff
@@ -1 +1,135 @@
...
+      season = game[:season]
+      goals = game[:home_goals].to_i + game[:away_goals].to_i
+
+      goals_by_season[season] += goals
+      games_by_season[season] += 1
+    end
+
+    average_goals = {}
+
+    goals_by_season.each do |season, total_goals|
```
[the remark this answers is not in the document]
Comment (bkchilidawg) [reply lib/stat_tracker.rb]: paired with and closed

File: lib/stat_tracker.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (trpubz) [reply lib/stat_tracker.rb:32]: your new logic is much cleaner and simpler, really liked it

File: lib/stat_tracker.rb (hunk 2 above)
Comment (trpubz) [reply lib/stat_tracker.rb:40]: 1 line, like it

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,4 @@
+.vscode/
+.byebug_history
```
Comment (kameronk92) [inline .gitignore:2]: thanks for catching this

**Review by kameronk92 — APPROVED** [review]
Changes looks copasetic with what was discussed on tuple

**Review by trpubz — APPROVED** [review]

**Pull request merged**